src/goto-programs/goto_program.cpp

```diff
@@ -711,8 +713,26 @@ void goto_programt::instructiont::validate(
     DATA_CHECK(targets.empty(), "assign instruction should not have a target");
     break;
   case DECL:
+    DATA_CHECK_WITH_DIAGNOSTICS(
+      code.get_statement() == ID_decl,
+      "declaration instructions should contain a declaration statement",
+      source_location);
+    DATA_CHECK_WITH_DIAGNOSTICS(
+      !ns.lookup(to_code_decl(code).get_identifier(), table_symbol),
+      "declaring unknown symbol: " +
```

Comment:
These texts seem to sometimes follow the style of "this property should hold" and other times "this property was violated". I think we should choose the same one every time (and probably prefer the former).